cirq/google/engine/engine_job.py

```diff
@@ -365,6 +369,18 @@ def _raise_on_failure(job: quantum.types.QuantumJob) -> None:
     def __iter__(self) -> Iterator[study.TrialResult]:
         return iter(self.results())
 
+    # pylint: disable=function-redefined
+    @overload
+    def __getitem__(self, item: int) -> study.TrialResult:
```

Comment:
In practice, the client code will probably also want to know the number of results available so that it can decide which indices are valid, so we should probably also implement `__len__`.